specutils/tests/test_loaders.py

```diff
@@ -733,13 +747,98 @@ def test_wcs1d_fits_writer(tmpdir, spectral_axis):
     assert quantity_allclose(spec.spectral_axis, spectrum.spectral_axis)
     assert quantity_allclose(spec.spectral_axis, disp)
     assert quantity_allclose(spec.flux, spectrum.flux)
+    assert np.all(spec.mask == spectrum.mask)
+    if uncertainty is None:
+        assert spec.uncertainty is None
+    else:
+        assert quantity_allclose(spec.uncertainty.quantity, spectrum.uncertainty.quantity)
 
     # Read from HDUList
     hdulist = fits.open(tmpfile)
     spec = Spectrum1D.read(hdulist, format='wcs1d-fits')
     assert isinstance(spec, Spectrum1D)
     assert quantity_allclose(spec.spectral_axis, spectrum.spectral_axis)
     assert quantity_allclose(spec.flux, spectrum.flux)
+    assert np.all(spec.mask == spectrum.mask)
+    if uncertainty is None:
+        assert spec.uncertainty is None
+    else:
+        assert quantity_allclose(spec.uncertainty.quantity, spectrum.uncertainty.quantity)
+
+
+@pytest.mark.parametrize("spectral_axis",
+                         ['WAVE', 'FREQ', 'ENER', 'WAVN'])
+@pytest.mark.parametrize("with_mask", [False, True])
+@pytest.mark.parametrize("uncertainty",
+                         [None, StdDevUncertainty, VarianceUncertainty, InverseVariance])
+def test_wcs1d_fits_cube(tmpdir, spectral_axis, with_mask, uncertainty):
+    """Test write/read for Spectrum1D spectral cube with WCS spectral_axis."""
+    wlunits = {'WAVE': 'Angstrom', 'FREQ': 'GHz', 'ENER': 'eV', 'WAVN': 'cm**-1'}
+    # Header dictionary for constructing WCS
+    hdr = {'CTYPE1': spectral_axis, 'CUNIT1': wlunits[spectral_axis],
+           'CRPIX1': 1, 'CRVAL1': 1, 'CDELT1': 0.01,
+           'WCSAXES': 3, 'DISPAXIS': 0,
+           'CRPIX2': 38.0, 'CRPIX3': 38.0, 'CRVAL2': 205.4384, 'CRVAL3': 27.004754,
+           'CTYPE2': 'RA---TAN', 'CTYPE3': 'DEC--TAN', 'CUNIT2': 'deg', 'CUNIT3': 'deg',
+           'CDELT2': 3.61111097865634E-05, 'CDELT3': 3.61111097865634E-05,
+           'PC1_1': 1.0, 'PC12 ': 0, 'PC1_3': 0,
+           'PC2_1': 0, 'PC2_2': -1.0, 'PC2_3': 0,
+           'PC3_1': 0, 'PC3_2': 0, 'PC3_3': 1.0}
+    # Create a small data set
+    flux = np.arange(1, 121).reshape((10, 4, 3))**2 * 1.e-14 * u.Jy
+    wlu = u.Unit(hdr['CUNIT1'])
+    wl0 = hdr['CRVAL1']
+    dwl = hdr['CDELT1']
+    disp = np.arange(wl0, wl0 + (flux.shape[2] - 0.5) * dwl, dwl) * wlu
+    tmpfile = str(tmpdir.join('_tst.fits'))
+
+    if with_mask:
+        die = np.random.Generator(np.random.MT19937(23))
+        mask = die.choice([0, 0, 0, 0, 0, 1], size=flux.shape).astype(np.uint16)
```

Comment:
Isn't it simpler to start with all zeroes and then assign select slices to ones? Is a random number generator really necessary? Looks cool though.

Reply:
😺 Maybe `5 * [0] + [1]` would be cooler still, but I did not really care as long as it fits on one line.
Do you mean is it necessary to use random numbers at all – probably not, just wanted to make up some more "realistic" example? Or to use `Generator` – that is how the `np.random` docs recommend it now.